File: samples/test/cross_lang/selection_sort/SelectionSort.c
#include <stdio.h>

int *selectionSort(int array[],int n)
{
	int j, temp, i, min, pos;
	i = 0;
	while (i < n)
	{
		min = array[i];
		pos = i;
		for (j = i + 1; j < n; j++) 
		{
			if (array[j] < min)
			{
				min = array[j];
				pos = j;
			}
		}
		temp = array[i];
		array[i] = array[pos];
		array[pos] = temp;
		i++;
	}
}

File: samples/test/cross_lang/selection_sort/SelectionSort.java
public class SelectionSort {
	public static void sort(int[] array) {
		int i, j, min, pos, temp;
		i = 0;
		while (i < array.length) {
			min = array[i];
			pos = i;
			for (j = i + 1; j < array.length; j++) {
				if (array[j] < min) {
					min = array[j];
					pos = j;
				}
			}
			temp = array[i];
			array[i] = array[pos];
			array[pos] = temp;
			i++;
		}
	}
}